코드 리뷰를 위한 PR

BadaHertz52 opened the pull request:
MVC 패턴을 적용해 기능을 구현했습니다.

파일 구조, 파일에 대한 설명과 순서도를 [root/README.md](https://github.com/BadaHertz52/javascript-christmas-6-BadaHertz52/blob/c56b25cba103175622b8c1825514e62f7aabb17e/README.md)에 첨부했으니 참고하시면 코드 리뷰시 도움이 될 것 같습니다.

BadaHertz52 committed:
chore<gitignore>: eslint, prettier관련 파일들을 gitignore에 추가

BadaHertz52 committed:
chore: docs/README.md 작성

- 크리스마스 이벤트 기능 구현 목록 추가
- 크리스 마스 이벤트의 도메인 로직, UI로직 내용 추가

BadaHertz52 committed:
chore: docs/README.md 작성

- 크리스마스 이벤트 기능 구현 목록 추가
- 크리스 마스 이벤트의 도메인 로직, UI로직 내용 추가

BadaHertz52 committed:
Merge branch 'main' of https://github.com/BadaHertz52/javascript-christmas-6-BadaHertz52 into main

BadaHertz52 committed:
docs:  상수 파일 생성 , 유틸 함수 파일 생성

- src 하우에 constants 폴더 , utils 폴더 생성
- 상수 파일 생성 : Menu, Message,Rule.index
- 유틸 함수 생성 : Freeze, MessageFactory,index

BadaHertz52 committed:
feat<OutPutView> : 파일 이동 및 메서드 추가

- src 하위에 views 폴더 생성, 여기로 이동
- 메서드 추가

BadaHertz52 committed:
docs<Message>: QUERY  속성명 변경

- reservation -> reservationDate
- menu-> order

BadaHertz52 committed:
docs<constants>: RegexPattern.js 생성

BadaHertz52 committed:
feat<InputView>: InputView 파일 위치 변경 및 메서드 추가

- src-> src/views 로 위치 변경
- InputView에 메서드 추가

BadaHertz52 committed:
feat<utils>: Validate.js 생성

- isSuitableForNumber 함수 생성

BadaHertz52 committed:
docs<constants, utils>: RegexPattern , Message 수정 및  getOutputHeader 삭제

- RegexPattern : NUMBER_REGEX_PATTERN -> DATE_REGEX_PATTERN 으로 변경, 정규 패턴 수정 및 :FOOD_ORDER_PATTERN 추가
-Message : -ERROR_MESSAGE.reservation 의 type 삭제 , ERROR_MESSAGE.menu의 wrongType 을 wrongFormat으로 수정 및 OUTPUT_HEADR_MESSAGE 수정

BadaHertz52 committed:
refactor: OutputView에 있는 예약방문일, 메뉴 주문 시 오류 메세지를 생성하는 코드를 함수 형태로 MessageFactory로 옮김

BadaHertz52 committed:
fix<InputView> : return 오류 수정

BadaHertz52 committed:
feat<reservation>: 예약 방문일에 대한 입력값 받기 및 유효성 검사 관련 기능 추가

-utils/Validate.js  :isSuitableForNumber -> testRegExp 으로 이름 변경 ,isSuitableForDate 추가

- App/js : getReservationDate 메서드 , reservation 속성 추가

- models/ReservationDate.js , models/index.js 생성

-controllers/index.js, controllesr/InputController.js 생성

BadaHertz52 committed:
fix: MessageFactory에 "}" 가 삭제된 오류 수정

BadaHertz52 committed:
feat: CustomError 추가 및 적용

-ReservationDate.js에 CustomError 적용

BadaHertz52 committed:
feat<order>: 메뉴 주문 형식에 대한 유효성 검사

- Message.js : noMenuDelimiter 삭제

- RegexPattern.js : FOOD_ORDER_PATTERN -> FOOD_ORDER_REGEX_PATTERN으로 이름 수정, 값 수정

- utils: testRegExp 이동 (Validate.js ->RegExpFactory.js) , Validate.js 삭제 , isSuitableForDate는 ReservationDate로 이동

- InputController : getValidOrder 메서드 추가

-Order.js 생성

-App : getOrder 메서드 추가

BadaHertz52 committed:
docs<MessageFactory> : getMenuErrorMessage에 대한 설명 수정

BadaHertz52 committed:
docs<OutputView>: printReservationErrorMessage, printMenuErrorMessage 삭제

BadaHertz52 committed:
Merge branch 'fn/error' into fn/order

BadaHertz52 committed:
feat<order> : 주문 형식에 최소 메뉴 개수도 포함시킴

- RegexPattern.js : FOOF_ORDER_REGEX_PATTERN이 메뉴 개수가 1~20 인 것도 검사하도록 수정

- Message.js : ERROR_MESSAGE.menu 에서 minNumber를 삭제하고 해당 내용을 wrongOrderFormat에 합침

-MessageFactroy.js: getMenuErrorMessage 의 param에 대한 설명 수정

BadaHertz52 committed:
feat<order>: 총 주문 개수에 대한 유효성 검사 추가

-OrderMenu.js 생성
-Menu.js : MENUS 수정 (fresszeFood의 paramter 수정)
-Messag.js : ERROR_MESSAGE.menu 의 wrongOrderFormat에 주문 최대 개수 추가
- Order.js : 총 주문 개수에 대한 유효성 검사를 진행하는 메서드 추가 및 주문 문자열에 대한 테스트와 대상을 분리
- RegexFactory.js : 파일명 수정 (RegExpFactory->RegexFactory)
-RegexPattern.js : 주문에 대한 정규표현식이 주문 최대 개수에 대해 유연하도록 수정

BadaHertz52 committed:
fix<InputView>- 입력값에 공백이 있을 경우, 이를 제거

BadaHertz52 committed:
feat<order>: 메뉴에 없는 음식 주문에 대한 기능 추가

-Message.js : 메뉴에 없는 음식 주문에 대한 오류 메세지 추가 ( ERROR_MESSAGE.menu에 none 추가 )
-MessageFactofy.js : getMenuErrorMessage에 대한 설명에서 params 에 "none"추가
-Order.js : orderedFoods , setOrderedFood, isInMenu 추가 및 Order 속성의 타입에 대한 설명 추가
-OrderedMenu : getFoodData -> getData 로 이름 변경

BadaHertz52 committed:
feat<order>: 주문 시 메뉴 중복 여부 검사하는 기능  추가

-Message.js : 메뉴 중복에 대한 오류 메세지 추가 및 코드 정리
-MessageFactory.js : getMenuErrorMessage의 param 정리 및 "duplicate"추가
-Order.js : isDuplicate 추가 및 주석 제거
-

BadaHertz52 committed:
feat<order>: 음료만 주문했는지에 대한 유효성 검사 기능, 유효한 주문 출력 기능 추가

- Order.js : 음료만 주문했는지에 대한 유효성 검사 메서드(hasOnlyBeverage) , list반환하는 메서드(getList)추가
- Menu.js : MENUS의 요소의  type을 영어로 통일
- InputController.js : Order의 list를 반환하도록 getValidOrder 수정
- OuputView.js : printOrder 수정
- App.js : getOrder, run 에 주문에 대한 코드 추가

BadaHertz52 committed:
docs: docs/README.md 수정 및 필요 없는 코드 삭제

-  docs/README.md : 현재 진행 상황에 맞추어 수정
- ReservationData.js ,Order.js : 불필요한 코드 삭제

BadaHertz52 committed:
docs: docs/README.md 수정 및 필요 없는 코드 삭제

-  docs/README.md : 현재 진행 상황에 맞추어 수정
- ReservationData.js : 불필요한 코드 삭제

BadaHertz52 committed:
feat<amount>: 할인 전 총 주문 금액 계산 및 출력 기능 추가

-  docs/README.md  :진행 상황에 맞추어업데이트
-  App.js : 할인 전 총 주문 금액 상태 업데이트 및 메서드 private로 수정
-  Rule.js : 변수명 변경 (MONEY_UNIT ->CURRENCY_UNIT)
-  Calculator.js, Money.js 추가
- Reservation.js: 변수명 변경에 따른 수정, 경로 오류 수정, 메서드 private로 수정
- OuputView.js : 빈줄을 출력하는 메서드 추가, printMoney ,printAmount 수정

BadaHertz52 committed:
feat<event> : 크리스마스 디데이, 평일할인 이벤트 적용 기능 추가

-   EventUtils.js 생성 : 이벤트 관련해 반복해서 사용하는 유틸 함수들 관리
- 이벤트 관련 모델 추가 :  XmasDDayEvent.js , WeekDayEvent.js
-  이벤트 컨트롤러 추가 : EventController.js
- 이벤트 관련 상수 추가 : EventRule.js
- THRESHOLD의 불필요한 속성 제거
-  ERROR_MESSAGA 의 range 수정 : 상수 사용
- App 의 속성명 수정 : totalAmount -> totalDiscount

BadaHertz52 committed:
feat<event> :  주말 할인 관련 기능 추가

-  주말 할인에 대한 모듈인  WeeekendEvent.js 추가
-  EventController.js 에 주말 할인 관련 메서드 추가

BadaHertz52 committed:
refactor: THRESOLD 중 이벤트와 관련있는 것들을   EventRule.js로 옮겨서 EVENT_THRESOLD 생성

BadaHertz52 committed:
feat<event>: 할인 전 총 금액이 이벤트 대상 금액 이상일 떼, 할인 금액에 대한 메서드들을 실행하도록 EventController,js 수정

BadaHertz52 committed:
refactor<order> : WEEK_DAY_EVENT,WEEKEND_EVENT 의 속성명 변경 (day-> days)

BadaHertz52 committed:
feat<event>: 특별 이벤트 관련 기능 추가

- SpecialEvent.js : 특별 이벤트에 대한 모듈
- EventController.js 에 특별 이벤트의 혜택을 가져오는 메서드 추가
- 이벤트 관련 모듈을 event/index.js에서 추출
- EventRule.js에서 필요없는 속성 삭제

BadaHertz52 committed:
feat<event> : 증정 여부에 대한 기능 추가

- GiftEvent.js 추가 : 증정에 대한 모델
- EventController.js : 증정에 대한 메서드 추가
- 증정품에 대한 상수(GIFT)추가

BadaHertz52 committed:
docs<error>: 오류 메세지에 요구 사항에 있는 문구대로 수정

- ERROR_MESSAGE 수정 :추가로 설정한 오류와 관련된 메세지 삭제 , menu 속성명을 order로 수정
- MessageFactory.js의 함수명, 함수 수정 : getReservationErrorMesage -> getReservationDateErrorMessage, getMenuErrorMessage-> getOrderErrroMesage
- Order.js : makeError 메서드 추가

BadaHertz52 committed:
feat<output> : OutputController (데이터 출력 컨트롤) 생성 및 OutputView.js,App,js 수정

BadaHertz52 committed:
feat<event> : 이벤트 총 혜택금액 계산 및 출력

- GiftEvent.js : getGift의 반환값 수정
- EventController.js : 메서드 이름 변경. benefit(Object)->benefits(Array) 속성명, 속성 값 형식 변경, 적용된 이벤트가 benefits에 추가되도록 수정
- Calculator.js: 이벤트 총 혜택 계산하는 메서드(calculatorBenefitAmount) 추가
- OUTPUT_HEADER_MESSAGE 에 gift 추가, benefitDetails->benefits으로 속성명 변경
- EVENT_NAMES 추가
- App.js : 이벤트 혜택과 총 혜택금액에 관한 메서드 추가 및 benefit 속성 제거, reservation에 totalBenefitAmount 추가ㅣ

BadaHertz52 committed:
docs: 경로 수정

- RegexPattern.js: 상수 가져오는 경로 수정

BadaHertz52 committed:
fix<event> : 이벤트 혜택, 혜택 금액이 없을 경우에 대한 기능 추가 및 수정

-이벤트 혜택이  없을 경우 "없음" 을 출력하는 기능 추가
-출력하는 금액이 0원일때 할인 기호가 붙지 않게 수정
- 상수명 변경 : NOE_EVENT_BENEFIT -> NONE

BadaHertz52 committed:
feat<event>: 할인 후 예상 결제 금액 계산 및 출력 기능 추가

- App.js : 할인 후 예상 결제 금액에 대한 메서드 추가
- Calculator.js : 할인 후 예상 결제 금액 계산하는 메서드 추가

BadaHertz52 committed:
chore: 진행 상황에 맞추어 docs/README.md 업데이트

BadaHertz52 committed:
Merge branch 'fn/event' into develop

# Conflicts:
#	docs/README.md
#	src/models/ReservationDate.js

BadaHertz52 committed:
refactore: EventRule.js의 상수 수정 및 freezeObejct 사용

- BADGES -> BAGE_SHAPE 로 변수명 변경, name.icon이라는 하위 속성 추가

BadaHertz52 committed:
refactore: EventRule.js의 상수 수정 및 freezeObejct 사용

- BADGES -> BAGE_SHAPE 로 변수명 변경, name.icon이라는 하위 속성 추가

BadaHertz52 committed:
feat<badge> : 총 혜택 금액에 따라 배지 부여하는 기능 추가

- Badge.js 추가
- OutputController.js., OutputView.js : 배지를 출력하는 메서드 추가
- App.js : 총 혜택 금액에 따라 배지를 설정하고 출력하는 메서드와 코드 추가

BadaHertz52 committed:
Merge branch 'fn/event' into develop

# Conflicts:
#	src/constants/EventRule.js

BadaHertz52 committed:
refactor:  상수 중, freeze 되지 않은 것들 freezeObeject 사용해 수정

BadaHertz52 committed:
feat<order>: 주문  형식에 대한 검사와 메뉴의 개수에 대한 검사를 분리

-  Rule.js : 주문 할 수 있는 메뉴의 최소,최대에 대한 상수 추가 및 속성명 변경 ( numberOfMenu-> numberOfOrder)
- 음식 주문에  대한 정규식 패턴 수정
- Order.js : 메서드명 변경 (validateNumberOfMenu ->validateTotalNumberOfOrder) , validateNumberOfOrder 메서드 추가

BadaHertz52 committed:
feat : 이벤트 주의 사항에 대한 안내 추가

- Message.js: 이벤트 주의 사항에 대한 안내관련 메세지 추가
- OuputView.js : printEventTarget 삭제, printNotesOnEvents 추가 및 빈줄 출력 관련해서  printGreeting, printEventPriew 수정

BadaHertz52 committed:
feat<error> : 주문에 대한 오류 메세지에 어떤 오류인지에 대한 메세지를 추가

- Message.js : 반복되는 문구를 상수 변수로 관리, 주문 관련 오류를 상세하게 수정
- Order.js : makeError 메서드 수정
- MessageFactory.js : getOrderErrorMessage, getErrorMessage 수정

BadaHertz52 committed:
chore: 진행 상황에 맞추어 docs/README.md 업데이트

BadaHertz52 committed:
chore: docs/README.md에 테스트 관련 내용 추가 및 오타 수정

BadaHertz52 committed:
test: ApplicationTest.js에 날찌 예외 테스트에 날짜 입력값 추가

BadaHertz52 committed:
refactor : 테스트에 반복적으로 사용하는 함수들을 testUtils/index.js로 옮김

BadaHertz52 committed:
fix: 메세지 수정

- Money 클래스를 이용하지 않고 상수에서 총주문 금액 가져오는 방식으로 변경
- ERROR_MESSAGEA 수정 : 속성명 변경 (footer->reInput), 예약방문일에 대한 오류 메세지 상세하게 수정
- .MessageFactory.js  방문일에 대한 오류 메세지와 주문에 대한 오류 메세지의 형식이 달라서 getErrorMesaage 삭제하고 함수에서 필요한 메세지를 직접 생성하는 방향으로 수정

BadaHertz52 committed:
fix<order> : 주문 입력에 대한 정규표현식 수정

- 입력한 메뉴가 유효하냐는 다른 유효성 테스트에서 검증하고, 입력 시에는 영문과 숫자도 사용할 수 있게 수정

BadaHertz52 committed:
Merge branch 'develop' of https://github.com/BadaHertz52/javascript-christmas-6-BadaHertz52 into test

BadaHertz52 committed:
fix : 총 주문 개수에 대한 에러 메세지가 삭제된 오류 수정

BadaHertz52 committed:
test<order> : 주문에 대한 테스트 추가

- OrderTest.js 올바르지 않은 주문에 대한 예외 테스트, 유효한 주문 시 주문 리스트 반환 테스트 추가
- docs/README.md : 진행 상황에 따라 업데이트

BadaHertz52 committed:
test : 할인 전 총 주문 금액에 대한 테스트 추가

- CalculatorTest.js  금액에 대한 테스트를 진행하는 파일 생성 , 할인 전 총 주문 금액에 대한 테스트 추가
- 진행 상황에 따라 docs/README.md 수정

BadaHertz52 committed:
test<event>: 증정 이벤트에 대한 테스트 추가

-GifteTest.js 추가
- 진행 상황에 맞추어 docs/REDAME.md 수정
- 불필요한 코드 삭제

BadaHertz52 committed:
refactor: 테스트 파일들의 변수명 수정

BadaHertz52 committed:
test<event> : 할인 전 총 구매 금액을 기준으로 한 이벤트 적용에 대한 테스트 추가 및 testUtils.js 수정

-EventTest.js :  할인 전 총 구매 금액을 기준으로 한 이벤트 적용에 대한 테스트
- testUtils.js  에서 LINE_SPEPAARATOR 내보내기

BadaHertz52 committed:
refactor : 주문 리스트를 만드는 반복적인 코드를 함수로 재사용할 수 있게 수정

-testUtils/index.js 에 getOrderList (주문 사항이 담긴 문자배열로 주문 리스트를 반환하는 함수) 추가
- 주문 리스트를 사용하는 테스트 파일에서 getOrderList  사용

BadaHertz52 committed:
fix<event>: WeekDayEvent, WeekendEvent의 getDiscount 수정

- 해당 이벤트 대상인 메뉴가 주문 목록에 없을 경우도 if절 조건에 포함 시킴

BadaHertz52 committed:
refactor: testUtils/index.js 에 있는 함수들을 testUtils 하위의 mock.js, orderList.js로 옮김

BadaHertz52 committed:
fix : 특별 할인 적용 날짜 에 대한 변수 오류 수정

BadaHertz52 committed:
test<event>:크리스마스 디데이 할인, 평일 할인, 주말 할인, 특별 할인 이벤트에 대한 테스트 추가

- EventTest.js : 이벤트에 대한 테스트 추가
- testUtils : 테스트 시 필요한 날짜에 대한 함수,상수 추가
- docs/README.md : 이벤트 테스트 할일들 체크

BadaHertz52 committed:
test<event> : 할인 혜택이 없는 이벤트를 출력 대상에서 제외시키는 지에 대한 테스트 추가

BadaHertz52 committed:
refactor : 상수 변수명을 대문자로 변경

BadaHertz52 committed:
test<event> : 증정 이벤트 시, 할인 가격을 샴페인 가격으로 하는지에 대한 테스트 코드 추가

BadaHertz52 committed:
fix<event> : 할인 후 예상 결제 금액에 대한 메서드 수정

- Calculator.js : 할인 후 예상 결제 금액은 할인 전 구매 금액에서 할임 금액의 합계를 제한 금액으로 총 혜택 금액에는 증정품 가격이 포함되어 있어서 증정품을 받았을 경우 이를 제하도록 수정
- App.js : 혜택 중 증정품을 받은 것이 있는지 판단하는 메서드 추가

BadaHertz52 committed:
fix : npm으로 test 시, index.js의 경로를 읽어오지 못하는 오류 수정

- index.js가 아닌, 해당 메서드나 변수가 선언된 파일 경로로 수정

BadaHertz52 committed:
test<event> : 금액에 대한 테스트 추가

-  총 혜택 금액, 할인 후 예상 결제 금액에 대한 테스트 추가
- Calculator에 대한 테스트와 App.run 실행 시에 대한 테스트로 분리
- mock.js에 makedExpected 함수 추가

BadaHertz52 committed:
fix<badge> : 배지 이름과 배지 아이콘을 따로 출력하도록 수정

BadaHertz52 committed:
test<badge> :  총 구매 헤택 금액 별로 배지를 주는 지에 대한 테스트 추가

BadaHertz52 committed:
refactor: 테스트 파일 리팩토링

BadaHertz52 committed:
chore : 진행 상황에 맞추어 docs/README.md 업데이트

BadaHertz52 committed:
docs<badge> : 배지 출력 시, 추가로 출력되는 메세지 수정

- 수여된 배지가 있으면 이모지로 해당 배지를 보여주고, 없으면 눈사람 이모지를 보여주는 방식으로 수정

BadaHertz52 committed:
feat : 사용자에게 메뉴판을 보여주는 문구를 출력하는 기능 추가

- MENUS_BY_TYPE, getMenusByType : 메뉴를 종류별로 정리한 상수 추가 및 해당 상수를 만드는 메서드 추가
- OutputView.js, OuputContorller,js, App.js 메뉴판을 출력하는 메서드를 추가

BadaHertz52 committed:
reafactor: 이벤트 혜택에 증정품이 있는 지 확인하는 코드들이 반복되어서 이를 유틸 함수로 만듦, benefit->benefits로 변경

BadaHertz52 committed:
refactor<event> : benefits 타입을 undefined 또는 { event: EventType; discount: number }[] 로 통일 시킴

- EventController의 benefits 초기값은 undefined로 수정하고 , addBenefit에서 benefits 의 현재값에 따라 benefits 값을 업데이트하는 코드로 수정
- benefits 타입이 수정된 대로, OuputController.js 와 Calculator.js 수정

BadaHertz52 committed:
refactor: 줄바꿈, 함수 또는 변수에 대한 설명문 추가, 불필요한 코드

- freesedFoodType 삭제

BadaHertz52 committed:
refactor: Moeny 파일을 models-> controllers 폴더로 이동

BadaHertz52 committed:
refactor: CurtomError.js 파일을 modesl-> custromError 폴더로 옮기고 파일명을 index.js로 변경

BadaHertz52 committed:
refactor : Calculator을 클래스가 아닌 객체로 변경

BadaHertz52 committed:
docs :root/README.md 수정 및 해당 파일에 사용되는 이미지 파일들 추가

BadaHertz52 committed:
docs: root/README.md  ,docs/README.md 수정

- 파일 구조 관련 오류 및 줄바꿈 오류 수정

BadaHertz52 committed:
docs : root/REAME.md 에  실행 화면 추가 및 설치및 실행 코드 추가

BadaHertz52 committed:
docs : 폴더명 수정 (readImges -> readmeImages)

BadaHertz52 committed:
refactor:: 폴더 이동 및 리팩토링

- Calculator.js 폴더 이동 (controllers-> models)
- CustromError 파일명은 index.js -> CustromError.js로 변경 후 models 폴더로 이동
- CustromError의 throwError 삭제 대신 ErrorController 추가
- OutputController에 controlPrintError 메서드 추가
- ReservationDate에 date 속성 추가 및 getDate 메서드 추가 그에 따른 InputController의 getValidReservationDate수정
- testUtils에서 getRandomReservation 내보내기

BadaHertz52 committed:
test<resevation> : 예약 방문일에 대한 테스트 추가

BadaHertz52 committed:
docs: 리팩토링에 따른 root/REDME.md 수정

BadaHertz52 committed:
refactor : ErrorController 삭제, throw new CustromError(error) 형태로 수정

BadaHertz52 committed:
docs: REDME 이미지 경로 오류 수정

BadaHertz52 committed:
docs: README 의 내용 수정 및 추가

BadaHertz52 committed:
docs : README.md에 파일에 대한 설명 추가

ss0526100 commented on src/App.js:
```diff
@@ -1,5 +1,102 @@
+    totalBenefitAmount: undefined,
+    badge: undefined,
+  };
+  async run() {
```
run 함수가 조금 긴 것 같아요..! 데이터를 입력받는 부분(이 부분은 조금 더 나뉠 여지가 있을수도..), 데이터를 계산하는 부분, 데이터를 출력하는 부분으로 나누는 등,

비슷한 기능이나 비슷한 순서에 있는 함수들을 묶어서 관리하면 코드 가독성이 더 좋을 것 같아요!

ss0526100 commented on src/constants/EventRule.js:
```diff
@@ -0,0 +1,101 @@
+  discount: 1000,
+});
+
+export {
```
Airbnb Style에서 하나의 객체만 반환하는 경우에는 export default를 쓰는 것이 권장되어요! 참고하세요!!

ss0526100 commented on src/App.js:
```diff
@@ -1,5 +1,102 @@
 class App {
-  async run() {}
+  #reservation = {
+    date: undefined,
```
들어가야할 값의 형식을 미리 넣어주면 코드 가독성이 좋을 것 같아요

예를들어 datesk Amount 등에는 0을 넣는 방식처럼요!

BadaHertz52 replied:
값이 아직 미정이라 undefined로 하는게  좋을 것 같아요
말씀하신 부분은 주석을 첨부하는 게 형식을 명확하게 나타낼것 같아서 해당 방향으로 개선하면 될것 같아요

ss0526100 reviewed (commented):
약소한 코드 리뷰 남기고 갑니다! 4주동안 고생 많으셨습니다!!

BadaHertz52 committed:
refactor : constatns,utils의 index.js에서 export 하는 코드를 보다 간결하게 변경

BadaHertz52 committed:
refactor : 새로운 controller를 만들어서 App.run 코드를 보다 짧게 변경

- amountOfBeforeDIscount -> amountBeforeDiscount로 변경
- controller에 CalculatorController, ReservationController 추가
- 변수,파라미터의 타입에 대한 설명 추가

BadaHertz52 committed:
docs: 변수, 파라미터,리턴값등 에 대한 타입 설명 추가

BadaHertz52 committed:
docs: 리팩토링에 따른 root/README.md 수정 및 App.js 의 빈줄 삭제

BadaHertz52 committed:
refactor: App 에 관련된 기능끼리 묶는 메서드를 추가, 불필요한 줄 삭제 및 생성함수,필드,메서드 사이에 공백 추가

- App의 run 함수의 길이를 줄이고자, 예약,할인 전 구매액, 할인 혜택, 할인 후 금액, 배지에 대한 값을 가져오는 함수과 이를 메세지로 출력하는 함수들을 묶는 메서드들을 추가
